src/submodule.c

```diff
@@ -1854,6 +1854,14 @@ static void submodule_release(git_submodule *sm)
 	git__free(sm);
 }
 
+git_submodule* git_submodule_dup(git_submodule *sm)
+{
+	if (!sm)
+		return;
```

Comment:
I guess you meant to return either `sm` or, explicitly, `NULL`?

Reply:
Sure, which one is more appropriate (in current codebase)?

Reply:
I think that would be `NULL`.

Reply:
I would emulate the existing `dup` functions like `git_commit_dup` and return an `int` and use an out parameter.  This way we can change the implementation later in case we add some failure case.

Reply:
@ethomson 

Should I just use `git_object_dup` function?

```c
int git_submodule_dup(git_submodule **out, git_submodule *source) {
  return git_object_dup((git_object **)out, (git_object *)source);
}
```